### PR TITLE
Nceplibs bufr poetry new

### PR DESCRIPTION
**Pull request by randytpierce**
This is a new PR for the work to resolve issue https://github.com/NOAA-GSL/VxIngest/issues/358
This code provides a third_party directory, an NCEPLIBS-bufr entry there and a build script that will download and build NCEPLIBS-bufr and create a python wheel for the python NCEPLIBS-bufr wrapper using poetry.

### CODE REVIEW

**Review by ian-noaa — COMMENTED** [review]
A couple of nits on the script & README. But otherwise looks good.

File: third_party/NCEPLIBS-bufr/build.sh
[the remark this answers is not in the document]
Comment (randytpierce) [reply third_party/NCEPLIBS-bufr/build.sh:1]: Good idea, I turned on the vscode spellchecker for shellscripts and it did notice a couple of things in comments.

File: third_party/NCEPLIBS-bufr/build.sh [hunk 1]
```diff
@@ -0,0 +1,168 @@
+#!/bin/bash
+
```
[the remark this answers is not in the document]
Comment (randytpierce) [reply third_party/NCEPLIBS-bufr/build.sh:2]: good idea I added the defaults ${1-} and ${2-} so it won't complain

File: third_party/NCEPLIBS-bufr/build.sh [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+# NCEPLIBSbufr_version="12.0.1"
+# Use a specific sha for now - we know that one builds
+NCEPLIBSbufr_version="0d9834838df19879d5469c4c121f81d00eb13c66"
+tmp_workdir=$(mktemp -d)
+cd $tmp_workdir
+#wget https://github.com/NOAA-EMC/NCEPLIBS-bufr/archive/refs/tags/v${NCEPLIBSbufr_version}.tar.gz
+wget https://github.com/NOAA-EMC/NCEPLIBS-bufr/archive/0d9834838df19879d5469c4c121f81d00eb13c66.zip
+#tar -xzf v${NCEPLIBSbufr_version}.tar.gz
+unzip 0d9834838df19879d5469c4c121f81d00eb13c66.zip
+cd NCEPLIBS-bufr-${NCEPLIBSbufr_version}
```
[the remark this answers is not in the document]
Comment (randytpierce) [reply third_party/NCEPLIBS-bufr/build.sh:87]: agreed

File: third_party/NCEPLIBS-bufr/ncepbufr/README.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+
+This is the mac workflow.
+[mac workflow](https://github.com/NOAA-EMC/NCEPLIBS-bufr/blob/develop/.github/workflows/MacOS.yml)
+
+There are other workflows i.e. linux, intel, etc.
+[workflows](https://github.com/NOAA-EMC/NCEPLIBS-bufr/blob/develop/.github/workflows)
+
+To download ...
+[download](https://github.com/NOAA-EMC/NCEPLIBS-bufr/archive/refs/heads/develop.zip)
+[download](https://github.com/NOAA-EMC/NCEPLIBS-bufr/archive/refs/tags/v${version}.tar.gz)
```
[the remark this answers is not in the document]
Comment (randytpierce) [reply third_party/NCEPLIBS-bufr/ncepbufr/README.md:40]: sure, the link should have the bufrversion variable and a little text to explain the version etc.

File: third_party/NCEPLIBS-bufr/ncepbufr/README.md [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+cmake -DCMAKE_INSTALL_PREFIX=./install -DENABLE_PYTHON=ON ..
+make -j2 VERBOSE=1
+make install
+ctest --verbose --output-on-failure --rerun-failed
+if [ $? -ne 0 ]
+then
+    echo "ctest did not pass!"
+    exit 1
+fi
+```
```
[the remark this answers is not in the document]
Comment (randytpierce) [reply third_party/NCEPLIBS-bufr/ncepbufr/README.md:114]: good call

**Pull request merged**